### PR TITLE
Feature/ylp 710 accept generic notifications bis

### PR DESCRIPTION
**Pull request by ydrisR**
Merge accept invite routes for teams. No need to have 2 dinstinct routes

* userID is already provided in the token, not need to provide it in the url
* teamId is already part of the confirm record (in the db)

### CODE REVIEW

File: api/invite_test.go [hunk 1]
```diff
@@ -673,7 +673,7 @@ func TestInviteResponds(t *testing.T) {
 		{
 			desc:     "valid request to accept an team invite",
 			method:   http.MethodPut,
-			url:      fmt.Sprintf("/accept/team/invite/%s/%s", "123.456.789", "123456"),
+			url:      fmt.Sprintf("/accept/team/invite"),
```
Comment (CedricPerotto) [inline api/invite_test.go]: No need to use Sprintf without variables (I didn't suggest all other usage). 
```suggestion
			url:      "/accept/team/invite",
```

File: api/invite.go [hunk 2]
```diff
@@ -387,49 +387,32 @@ func (a *Api) AcceptInvite(res http.ResponseWriter, req *http.Request, vars map[
...
-//Accept the team invite for a member
+//Accept the given invite
 //
 // http.StatusOK when accepted
 // http.StatusBadRequest when the incoming data is incomplete or incorrect
 // http.StatusForbidden when mismatch of user ID's, type or status
 // @Summary Accept the given invite
 // @Description  This would be PUT by the web page at the link in the invite email. No authentication is required.
-// @ID hydrophone-api-acceptTeamInvite
+// @ID hydrophone-api-acceptAnyInvite
```
Comment (franck-fourel) [inline api/invite.go]:
```suggestion
// @ID hydrophone-api-acceptTeamNotifs
```

File: api/invite.go [hunk 3]
```diff
@@ -464,14 +447,13 @@ func (a *Api) AcceptTeamInvite(res http.ResponseWriter, req *http.Request, vars
...
-	} else {
-		conf.ValidateType(models.TypeMedicalTeamPatientInvite, &validationErrors)
-	}
+		ValidateType([]models.Type{
+			models.TypeMedicalTeamDoAdmin,
+			models.TypeMedicalTeamRemove,
+			models.TypeMedicalTeamInvite,
+			models.TypeMedicalTeamPatientInvite,
+		}, &validationErrors).
+		ValidateUserID(inviteeID, &validationErrors)
```
Comment (CedricPerotto) [inline api/invite.go:456]: We don't have to validate teamID any more ? 
( `ValidateTeamID(teamID, &validationErrors)` has been removed).

In that case ValidateTeamID method is not used anywhere.

File: api/invite.go [hunk 4]
```diff
@@ -485,14 +467,42 @@ func (a *Api) AcceptTeamInvite(res http.ResponseWriter, req *http.Request, vars
 		return
 	}
 
+	switch conf.Type {
+	case models.TypeMedicalTeamPatientInvite:
+		a.acceptTeamInvite(res, req, conf)
+	case models.TypeMedicalTeamInvite:
+		a.acceptTeamInvite(res, req, conf)
```
Comment (CedricPerotto) [inline api/invite.go]: I think we can use a case list in go 
```suggestion
	case models.TypeMedicalTeamPatientInvite, models.TypeMedicalTeamInvite :
		a.acceptTeamInvite(res, req, conf)
```

File: api/invite.go [hunk 5]
```diff
@@ -1011,7 +1021,7 @@ func (a *Api) SendTeamInvite(res http.ResponseWriter, req *http.Request, vars ma
 }
 
 // @Summary Send notification to an hcp that becomes admin
-// @Description  Send an email and a notification to the new admin user. Removing the admin role does not trigger any notification or email.
+// @Description  Send an email and a notification to the new admin user. The role change is done but notification is pushed for information (notifacation status is set to pending). Removing the admin role is managed as an exception. It does not trigger any notification or email.
```
Comment (CedricPerotto) [inline api/invite.go]:
```suggestion
// @Description  Send an email and a notification to the new admin user. The role change is done but notification is pushed for information (notification status is set to pending). Removing the admin role is managed as an exception. It does not trigger any notification or email.
```

**Review by CedricPerotto — CHANGES_REQUESTED** [review]

File: api/invite_test.go (hunk 1 above)
Comment (franck-fourel) [inline api/invite_test.go]:
```suggestion
			url:      "/accept/team/invite",
```

File: api/invite_test.go [hunk 6]
```diff
@@ -683,7 +683,7 @@ func TestInviteResponds(t *testing.T) {
 		{
 			desc:     "valid request to accept an team invite for a patient",
 			method:   http.MethodPut,
-			url:      fmt.Sprintf("/accept/team/invite/%s/%s", "123.456.789", "123456"),
+			url:      fmt.Sprintf("/accept/team/invite"),
```
Comment (franck-fourel) [inline api/invite_test.go]:
```suggestion
			url:      "/accept/team/invite",
```

File: api/invite_test.go [hunk 7]
```diff
@@ -692,19 +692,19 @@ func TestInviteResponds(t *testing.T) {
 			},
 		},
 		{
-			desc:     "not authorized request to accept an team invite",
+			desc:     "not authorized request to accept a team invite",
 			method:   http.MethodPut,
-			url:      fmt.Sprintf("/accept/team/invite/%s/%s", "not.authorized", "123456"),
+			url:      fmt.Sprintf("/accept/team/invite"),
```
Comment (franck-fourel) [inline api/invite_test.go]:
```suggestion
			url:      "/accept/team/invite",
```

File: api/invite_test.go [hunk 8]
```diff
@@ -692,19 +692,19 @@ func TestInviteResponds(t *testing.T) {
...
 			body: testJSONObject{
-				"key": "medicalteam.invite.member",
+				"key": "medicalteam.invite.wrong.member",
 			},
 		},
 		{
 			desc:     "invitation does not exist",
 			method:   http.MethodPut,
-			url:      fmt.Sprintf("/accept/team/invite/%s/%s", "123.456.789", "123456"),
+			url:      fmt.Sprintf("/accept/team/invite"),
```
Comment (franck-fourel) [inline api/invite_test.go]:
```suggestion
			url:      "/accept/team/invite",
```

File: api/invite_test.go [hunk 9]
```diff
@@ -714,13 +714,81 @@ func TestInviteResponds(t *testing.T) {
 		{
 			desc:     "invalid invitation",
 			method:   http.MethodPut,
-			url:      fmt.Sprintf("/accept/team/invite/%s/%s", "123.456.789", "123456"),
+			url:      fmt.Sprintf("/accept/team/invite"),
```
Comment (franck-fourel) [inline api/invite_test.go]:
```suggestion
			url:      "/accept/team/invite",
```

**Review by franck-fourel — CHANGES_REQUESTED** [review]
couple of comments

File: api/invite.go (hunk 4 above)
Comment (ydrisR) [reply api/invite.go]: cool :+1:

File: api/invite.go (hunk 3 above)
Comment (CedricPerotto) [reply api/invite.go:456]: My point was also that we don't check anymore that conf.TeamID is not nil (or accept.TeamID not sure if we have to check that in the payload or in the db response).

**Review by CedricPerotto — APPROVED** [review]

**Pull request merged**